src/aggregate/aggregate_plan.c

```diff
@@ -138,6 +138,22 @@ PLN_ArrangeStep *AGPLN_GetArrangeStep(AGGPlan *pln) {
   return NULL;
 }
 
+PLN_ArrangeStep *AGPLN_AddKNNArrangeStep(AGGPlan *pln, size_t k, const char *distFieldName) {
```

Comment:
consider making this function more general (mostly rename to something like `AGPLN_AddArrangeStepByField` and give it `limit` instead of `k`)